echo-notifications/src/main/groovy/com/netflix/spinnaker/echo/slack/SlackAttachment.groovy

```diff
@@ -38,9 +45,60 @@ class SlackAttachment {
   // The pretty date will appear in the footer
   long ts = System.currentTimeMillis() / 1000
 
-  public SlackAttachment(String title, String text, String color = '#cccccc') {
+  SlackAttachment(String title, String text, String color = '#cccccc') {
     this.title = title
     this.text = this.fallback = text
     this.color = color
+    this.actions = actions
+  }
+
+  SlackAttachment(String title, String text, InteractiveActions interactiveActions) {
+    this(title, text)
+
+    if (interactiveActions != null) {
+      this.actions = extractSlackActions(interactiveActions)
+      this.callback_id = interactiveActions.callbackServiceId + ":" + interactiveActions.callbackMessageId
+      this.color = interactiveActions.color
+    }
+  }
+
+  private List<SlackAction> extractSlackActions(InteractiveActions interactiveActions) {
+    List<SlackAction> slackActions = interactiveActions.actions.stream()
+      .filter() { it instanceof ButtonAction }
+      .map { SlackButtonAction.copy(it) }
+      .collect(Collectors.toList())
+    slackActions
+  }
+}
+
+class SlackAction {
```

Comment:
Is this class immutable? If so, maybe `final` on the properties?

Reply:
I wasn't 100% sure how Jackson works with Groovy classes. Will give it a try.